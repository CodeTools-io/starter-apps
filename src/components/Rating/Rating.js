import React from 'react';
import { Box } from 'grommet';
import { Star } from 'grommet-icons';
export default function Rating({ total = 5, selected = 0 }) {
  const stars = Array(total).fill(Star);

  return (
    <Box direction="row">
      {stars.map((Star, index) => {
        if (index < selected) {
          return <Star key={`rating-star-${index}`} color="#FFC95E" />;
        }

        return <Star key={`rating-star-${index}`} />;
      })}
    </Box>
  );
}
