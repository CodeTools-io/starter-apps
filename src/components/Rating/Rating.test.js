import React from 'react'
import Rating from './Rating'

describe('Rating', () => {
  test('can render', () => {
    const { container } = renderComponent(
      <Rating>
        <p>Content for the Rating</p>
      </Rating>
    )

    expect(container).toMatchSnapshot()
  })
})

