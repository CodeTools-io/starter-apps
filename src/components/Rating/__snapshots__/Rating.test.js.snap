// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating can render 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 bIdaiR"
  >
    <svg
      aria-label="Star"
      class="StyledIcon-ofa7kd-0 jXwebn"
      viewBox="0 0 24 24"
    >
      <polygon
        fill="#FFC95E"
        fill-rule="evenodd"
        points="12 16.667 5 22 8 14 2 9.5 9.5 9.5 12 2 14.5 9.5 22 9.5 16 14 19 22"
      />
    </svg>
    <svg
      aria-label="Star"
      class="StyledIcon-ofa7kd-0 jXwebn"
      viewBox="0 0 24 24"
    >
      <polygon
        fill="#FFC95E"
        fill-rule="evenodd"
        points="12 16.667 5 22 8 14 2 9.5 9.5 9.5 12 2 14.5 9.5 22 9.5 16 14 19 22"
      />
    </svg>
    <svg
      aria-label="Star"
      class="StyledIcon-ofa7kd-0 jXwebn"
      viewBox="0 0 24 24"
    >
      <polygon
        fill="#FFC95E"
        fill-rule="evenodd"
        points="12 16.667 5 22 8 14 2 9.5 9.5 9.5 12 2 14.5 9.5 22 9.5 16 14 19 22"
      />
    </svg>
    <svg
      aria-label="Star"
      class="StyledIcon-ofa7kd-0 jXwebn"
      viewBox="0 0 24 24"
    >
      <polygon
        fill="#FFC95E"
        fill-rule="evenodd"
        points="12 16.667 5 22 8 14 2 9.5 9.5 9.5 12 2 14.5 9.5 22 9.5 16 14 19 22"
      />
    </svg>
    <svg
      aria-label="Star"
      class="StyledIcon-ofa7kd-0 jXwebn"
      viewBox="0 0 24 24"
    >
      <polygon
        fill="#FFC95E"
        fill-rule="evenodd"
        points="12 16.667 5 22 8 14 2 9.5 9.5 9.5 12 2 14.5 9.5 22 9.5 16 14 19 22"
      />
    </svg>
  </div>
</div>
`;
